src/napari_experimental/group_layer.py

```diff
@@ -180,6 +241,88 @@ def _add_new_item(
                 group_items = ()
             insertion_group.insert(insertion_index, GroupLayer(*group_items))
 
+    def _move_plan(
+        self, sources: Iterable[NestedIndex], dest_index: NestedIndex
+    ):
+        """Prepare indices for a multi-move.
+
+        Given a set of ``sources`` from anywhere in the list,
+        and a single ``dest_index``, this function computes and yields
+        ``(from_index, to_index)`` tuples that can be used sequentially in
+        single move operations.  It keeps track of what has moved where and
+        updates the source and destination indices to reflect the model at each
+        point in the process.
+
+        This is useful for a drag-drop operation with a QtModel/View.
+
+        Parameters
+        ----------
+        sources : Iterable[NestedIndex]
+            An iterable of NestedIndex that should be moved to ``dest_index``.
+        dest_index : Tuple[int]
+            The destination for sources.
+        """
+        if isinstance(dest_index, slice):
+            raise TypeError(
+                trans._(
+                    "Destination index may not be a slice",
+                    deferred=True,
+                )
+            )
+
+        to_move: List[NestedIndex] = []
+        for idx in sources:
+            if isinstance(idx, tuple):
+                to_move.append(idx)
+            elif isinstance(idx, int):
+                to_move.append((idx,))
+            elif isinstance(idx, slice):
+                to_move.extend(list(range(*idx.indices(len(self)))))
+                raise NotImplementedError("Should never hit this case?")
```

Comment:
Maybe just remove checking for slice? Or make the `NotImplementedError` a bit more descriptive e.g. 'Slices can't be used in sources'